helpers/cache_helper.py

```diff
@@ -1,16 +1,34 @@
 import json
+import logging
 import os
 
 
 class Cache:
     def __init__(self, cache_file):
         self.cache_file = cache_file
-        self.data = {}
+        self.data: CacheDict = CacheDict()
+        self.data.dirty = False
+        self.log = logging.getLogger(__name__)
+        self.log.info(f"Debug: self.data.dirty = {self.data.dirty}")
 
         if os.path.exists(self.cache_file):
             with open(self.cache_file, "r") as f:
                 self.data = json.load(f)
 
     def save(self):
-        with open(self.cache_file, "w") as f:
-            json.dump(self.data, f)
+        if self.data.dirty:
+            self.log.info("DEBUG: cache is dirty, saving")
+            with open(self.cache_file, "w") as f:
+                json.dump(self.data, f)
+
+            self.data.dirty = False
+
+
+# Use this for the cache so we can mark it dirty on any update
+class CacheDict(dict):
+    def __init__(self):
+        self.dirty = False
+
+    def __setitem__(self, k, v):
+        self.dirty = True
+        super(CacheDict, self).__setitem__(k, v)
```

Comment:
Wouldn't this be `super(dict, self).__setitem__(k,v)` ?

Reply:
Ah, you can just write this as `super().__setitem__(k,v)` but it seems it is the same as you have it here.